### PR TITLE
Fee related part 1

### PR DESCRIPTION
**Pull request by wangminqi**
-relocate pallet-payment-interface into parachain/pallets
-implement pallet-payment-interface in parachain runtime

### CODE REVIEW

**Comment by wangminqi** [comment]
Wait for CI improvement

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -99,12 +99,10 @@ pub fn get_chain_spec_dev(id: ParaId) -> ChainSpec {
...
-				vec![
-					(
-						get_account_id_from_seed::<sr25519::Public>("Alice"),
-						get_collator_keys_from_seed("Alice"),
-					),
-				],
+				vec![(
+					get_account_id_from_seed::<sr25519::Public>("Alice"),
+					get_collator_keys_from_seed("Alice"),
+				)],
```
Comment (Satoshi-Kusumoto) [inline node/src/chain_spec.rs]: better not to change

File: node/src/command.rs [hunk 2]
```diff
@@ -27,14 +27,16 @@ fn load_spec(
 	Ok(match id {
 		"dev" => Box::new(chain_spec::get_chain_spec_dev(para_id)),
 		"staging" => Box::new(chain_spec::get_chain_spec_staging(para_id)),
-		"generate-prod" => Box::new(chain_spec::get_chain_spec_prod(para_id)), // to generate res/chain_spec/prod.json
+		"generate-prod" => Box::new(chain_spec::get_chain_spec_prod(para_id)), /* to generate
+		                                                                         * res/chain_spec/
+		                                                                         * prod.json */
```
Comment (Satoshi-Kusumoto) [inline node/src/command.rs]: ditto

File: node/src/command.rs [hunk 3]
```diff
@@ -27,14 +27,16 @@ fn load_spec(
...
 		"" | "prod" => Box::new(chain_spec::ChainSpec::from_json_bytes(
 			&include_bytes!("../res/chain_spec/prod.json")[..],
 		)?),
 		path => {
 			let chain_spec = chain_spec::ChainSpec::from_json_file(path.into())?;
 			Box::new(chain_spec)
-		}
+		},
```
Comment (Satoshi-Kusumoto) [inline node/src/command.rs:39]: You runned fmt?

File: pallets/payment-interface/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Substrate FRAME Dependencies
+frame-support = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
+frame-system = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
+
+## Substrate Pallet Dependencies
+pallet-transaction-payment = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
+pallet-treasury = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
+pallet-authorship = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
+pallet-balances = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.11", default-features = false }
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/Cargo.toml]: order

File: pallets/payment-interface/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	"sp-core/std",
+	"sp-io/std",
+	"sp-runtime/std",
+	"sp-std/std",
+	"frame-support/std",
+	"frame-system/std",
+	"pallet-transaction-payment/std",
+	"pallet-treasury/std",
+	"pallet-authorship/std",
+	"pallet-balances/std",
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/Cargo.toml]: ditto

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -916,6 +940,9 @@ construct_runtime! {
 		CumulusXcm: cumulus_pallet_xcm::{Pallet, Call, Event<T>, Origin} = 52,
 		DmpQueue: cumulus_pallet_dmp_queue::{Pallet, Call, Storage, Event<T>} = 53,
 
+		// Litentry Pallet
+		PaymentInterface: pallet_payment_interface::{Pallet, Call, Storage, Config<T>, Event<T>} = 60,
+
```
Comment (Satoshi-Kusumoto) [inline runtime/src/lib.rs:916]: This should be token related

File: pallets/payment-interface/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct RatioOf {
+	treasury: u32,
+	author: u32,
+	burned: u32,
+}
+
+impl Default for RatioOf {
+	fn default() -> Self {
+		RatioOf { treasury: 0, author: 0, burned: 1 }
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/src/lib.rs]: Better to use 100 and write in documentation that recommendation is to set it sum up to 100
Also set initial value to treasury = 100

File: pallets/payment-interface/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,515 @@
...
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn get_ratio)]
+	pub type Ratio<T: Config> = StorageValue<_, RatioOf, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn fix_block_reward)]
+	pub type FixBlockReward<T: Config> = StorageValue<_, u32, ValueQuery>;
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/src/lib.rs]: Next time should not have it here

File: pallets/payment-interface/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,515 @@
...
+		RatioOverflow,
+		BlockRewardTooLow,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10)]
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/src/lib.rs:109]: Pls put weight related todos

File: pallets/payment-interface/src/lib.rs [hunk 10]
```diff
@@ -0,0 +1,515 @@
...
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// Reward meet minimum_balance requirement of account existence
+			if 0 < block_reward &&
+				BalanceOf::<T>::from(block_reward) <
+					<T::Currency as Currency<T::AccountId>>::minimum_balance()
+			{
+				return Err(Error::<T>::BlockRewardTooLow.into())
+			}
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/src/lib.rs]: As discussed don't need this

File: pallets/payment-interface/src/lib.rs [hunk 11]
```diff
@@ -0,0 +1,515 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate as pallet_payment_interface;
+
+	use std::cell::RefCell;
+
+	// use codec::Encode;
```
Comment (Satoshi-Kusumoto) [inline pallets/payment-interface/src/lib.rs]: ???

**Review by Satoshi-Kusumoto — CHANGES_REQUESTED** [review]

File: node/src/command.rs (hunk 3 above)
Comment (wangminqi) [reply node/src/command.rs:39]: yes. eeee. Should I?

File: node/src/chain_spec.rs (hunk 1 above)
Comment (wangminqi) [reply node/src/chain_spec.rs]: I think fmt does this change.

File: node/src/command.rs (hunk 2 above)
Comment (wangminqi) [reply node/src/command.rs]: fmt effect.

File: node/src/command.rs (hunk 3 above)
Comment (wangminqi) [reply node/src/command.rs:39]: This is fmt's change.

**Pull request closed without merging**